spec-tests/src/test/java/org/candlepin/spec/InstanceSpecTest.java

```diff
@@ -0,0 +1,203 @@
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class InstanceSpecTest {
+    private ApiClient adminClient;
+    private OwnerDTO owner;
+    private String ownerKey;
+
+    @BeforeEach
+    public void setup() {
+        adminClient = ApiClients.admin();
+        owner = adminClient.owners().createOwner(Owners.random());
+        ownerKey = owner.getKey();
+    }
+
+    @Test
+    public void shouldAutoSubscribePhysicalSystemsWithQuantity2PerSocketPair() {
+        ProductDTO engProd = adminClient.ownerProducts()
+            .createProductByOwner(ownerKey, Products.randomEng());
+
+        Set<ConsumerInstalledProductDTO> installedProds = Set.of(new ConsumerInstalledProductDTO()
+            .productId(engProd.getId())
+            .productName(engProd.getName()));
+
+        int cpuSockets = 8;
+        ConsumerDTO physicalUser = createPhysicalUser(adminClient, owner, installedProds, cpuSockets);
+        ApiClient physicalClient = ApiClients.trustedConsumer(physicalUser.getUuid());
+
+        String systemId = StringUtil.random("id");
+        GuestIdDTO guest = new GuestIdDTO();
+        guest.setGuestId(systemId);
+        physicalUser.setGuestIds(List.of(guest));
+        physicalUser.setReleaseVer(new ReleaseVerDTO().releaseVer(""));
+        physicalClient.consumers().updateConsumer(physicalUser.getUuid(), physicalUser);
+
+        ConsumerDTO guestUser = createGuestUser(adminClient, owner, installedProds, systemId);
+        ApiClient guestClient = ApiClients.trustedConsumer(guestUser.getUuid());
+
+        int socketCount = 2;
+        // How much the quant increases by per ent
+        int socketMultiplier = 2;
+        ProductDTO instanceProduct =
+            createInstanceProduct(adminClient, ownerKey, socketCount, socketMultiplier, engProd);
+
+        // The pool needs source subscription and upstream pool id in order for the subpool to be created
+        int poolQuantity = 20;
+        adminClient.owners().createPool(ownerKey, Pools.random(instanceProduct)
+            .quantity(Long.valueOf(poolQuantity))
+            .subscriptionId(StringUtil.random("sourceSub-"))
+            .upstreamPoolId(StringUtil.random("upstream-"))
+            .subscriptionSubKey("master"));
+
+        List<PoolDTO> pools =  adminClient.pools()
+            .listPoolsByOwnerAndProduct(owner.getId(), instanceProduct.getId());
+        assertThat(pools)
+            .hasSize(2)
+            .extracting("type")
+            .containsExactlyInAnyOrder("NORMAL", "UNMAPPED_GUEST");
+
+        assertThat(guestClient.pools().listPoolsByConsumer(guestUser.getUuid()))
+            .isNotNull()
+            .hasSize(1);
+
+        physicalClient.consumers().bindProduct(physicalUser.getUuid(), engProd.getId());
+
+        int expectedQuantity = ((int) Math.ceil((double) cpuSockets / socketCount) * socketMultiplier);
```

Comment:
The original test asserts against a hard coded quantity value of 8 which made understanding the test difficult, so I wanted to calculate the expected quantity for clarity.

I believe this is correct based on some of the logic I saw in the rules.js quantity calculation methods and some of the testing I did , but maybe @wottop can help verify?